react/src/components/organisms/GeneralTeaser/index.js

```diff
@@ -0,0 +1,252 @@
+    }
+  },
+  level: PropTypes.number,
+  date: PropTypes.string,
+  org: PropTypes.string,
+  // eslint-disable-next-line consistent-return
+  description: (props, propName, componentName) => {
+    const component = props[propName];
+    const isValid = (comp) => {
+      if (!comp) {
+        return false;
+      }
+      if (typeof comp.type === 'string') {
+        return comp.type === 'Paragraph';
+      }
+      return comp.type.name && comp.type.name === 'Paragraph';
+    };
+    if (component && !isValid(component)) {
+      return new Error(`Invalid prop ${propName} supplied to ${componentName}. Got: ${component.type.name}. Validation failed.`);
+    }
+  },
+  // eslint-disable-next-line consistent-return
+  subLinks: (props, propName, componentName) => {
+    const component = props[propName];
+    const isValid = (comp) => {
+      if (!comp) {
+        return false;
+      }
+      if (!Array.isArray(comp)) {
+        return false;
+      }
+      let valid = false;
+      comp.every((child) => {
+        if (!child) {
+          return valid;
+        }
+        if (typeof child.type === 'string') {
+          valid = child.type === 'DecorativeLink';
+        }
+        valid = child.type.name && child.type.name === 'DecorativeLink';
+        return valid;
+      });
+      return valid;
+    };
+    if (component && !isValid(component)) {
+      return new Error(`Invalid prop ${propName} supplied to ${componentName}. Got: ${component.type.name}. Validation failed.`);
+    }
+  },
+  // eslint-disable-next-line consistent-return
+  primaryInfo: (props, propName, componentName) => {
+    const component = props[propName];
+    const isValid = (comp) => {
+      if (!comp) {
+        return false;
+      }
+      if (typeof comp.type === 'string') {
+        return comp.type === 'ContactGroup';
+      }
+      return comp.type.name && comp.type.name === 'ContactGroup';
+    };
+    if (component && !isValid(component)) {
+      return new Error(`Invalid prop ${propName} supplied to ${componentName}. Got: ${component.type.name}. Validation failed.`);
+    }
+  },
+  // eslint-disable-next-line consistent-return
+  secondaryInfo: (props, propName, componentName) => {
+    const component = props[propName];
+    const isValid = (comp) => {
+      if (!comp) {
+        return false;
+      }
+      if (!Array.isArray(comp)) {
+        return false;
+      }
+      let valid = false;
+      comp.every((child) => {
+        if (!child) {
+          return valid;
+        }
+        if (typeof child.type === 'string') {
+          valid = child.type === 'IconLink';
+        }
+        valid = child.type.name && child.type.name === 'IconLink';
+        return valid;
+      });
+      return valid;
+    };
+    if (component && !isValid(component)) {
+      return new Error(`Invalid prop ${propName} supplied to ${componentName}. Got: ${component.type.name}. Validation failed.`);
```

Comment:
consider `throw` for these (Error is intended as an exception object)
removes the need for the `eslint-disable-next-line consistent-return` disables, too

Reply:
I think doing it this way way recommended by React docs instead of `throw`